### PR TITLE
Feature projects improvements

### PR DESCRIPTION
**Pull request by avoevodin**
* except of documentation, tutorial for ORM and hooks for edit and del forms.

### CODE REVIEW

File: brigadier/projects/templates/project_detail.html [hunk 1]
```diff
@@ -3,7 +3,7 @@
 {% load i18n %}
 
 {% block container %}
-    <h1>{{ project.project_name }}</h1>
+    <h1>{{ object.project_name }}</h1>
```
Comment (iliadmitriev) [inline brigadier/projects/templates/project_detail.html:6]: why do you changed context variable name from project to object?

File: brigadier/projects/templates/task_list_included.html [hunk 2]
```diff
@@ -0,0 +1,51 @@
+{% load i18n %}
+
+{% if task_list %}
```
Comment (iliadmitriev) [inline brigadier/projects/templates/task_list_included.html:3]: this check should be done before including a template

File: brigadier/projects/views.py [hunk 3]
```diff
@@ -1,17 +1,38 @@
...
     template_name = 'project_list.html'
-    ordering = 'deadline'
+
+    def get_queryset(self):
+        """todo()
+
+        """
+        return annotate_completed_percentage(
+            ['id', 'project_name', 'deadline', 'budget', 'closed']
+        ).order_by('deadline')
```
Comment (iliadmitriev) [inline brigadier/projects/views.py:35]: unnecessary refactoring

**Review by iliadmitriev — CHANGES_REQUESTED** [review]

File: brigadier/projects/templates/project_detail.html (hunk 1 above)
Comment (avoevodin) [reply brigadier/projects/templates/project_detail.html:6]: i've deleted model from view. And there wasn't variable project in my template as default.

File: brigadier/projects/templates/task_list_included.html (hunk 2 above)
Comment (avoevodin) [reply brigadier/projects/templates/task_list_included.html:3]: Why? In project detail view we could face with same situation when we haven't any task in the project.

File: brigadier/projects/views.py (hunk 3 above)
Comment (avoevodin) [reply brigadier/projects/views.py:35]: Why? It was changed to avoid duplicate code in detail and list views.

Comment (iliadmitriev) [reply brigadier/projects/views.py:35]: then it should be written completely different
don't try to shoot in your foot

File: brigadier/projects/templates/project_detail.html (hunk 1 above)
Comment (iliadmitriev) [reply brigadier/projects/templates/project_detail.html:6]: its because you've broken something

File: brigadier/templates/nav.html [hunk 4]
```diff
@@ -2,24 +2,44 @@
...
-                 <li class="nav-item">
-                    <a class="nav-link" href="{% url "employees:list" %}">{% translate "Employees" %}</a>
+                <li class="nav-item">
+                    {% if request.GET.nb == "Employees" %}
+                        <a class="nav-link active"
+                           href="{% url "employees:list" %}?nb=Employees">{% translate "Employees" %}</a>
+                    {% else %}
+                        <a class="nav-link"
+                           href="{% url "employees:list" %}?nb=Employees">{% translate "Employees" %}</a>
+                    {% endif %}
```
Comment (iliadmitriev) [inline brigadier/templates/nav.html:42]: you should rollback this file from master branch
and write code from very beginning
but this time without trying to shoot in a foot
this solution is very complicated and hardly maintainable
avoid additional parameters
use only current request path
```
{% if request.path == url ... %}active{% endif %}
```

**Review by iliadmitriev — CHANGES_REQUESTED** [review]
* simplify solution with navbar active link
* fix bug with editing and saving project from list view (it ends up with 404)

**Review by iliadmitriev — APPROVED** [review]

File: brigadier/projects/views.py [hunk 5]
```diff
@@ -57,7 +100,15 @@ class ProjectEditView(generic.UpdateView):
...
-    success_url = reverse_lazy('projects:list')
+
+    def get_success_url(self):
+        """todo()
+
+        """
+        if self.request.GET.get('next'):
+            return self.request.GET.get('next')
+        else:
+            reverse_lazy('projects:list')
```
Comment (iliadmitriev) [inline brigadier/projects/views.py:111]: should return value,
in this case you should not use lazy

**Review by iliadmitriev — CHANGES_REQUESTED** [review]
fix bug

**Review by iliadmitriev — APPROVED** [review]

**Review by iliadmitriev — APPROVED** [review]

**Pull request merged**